server/controllers/temporary-accommodation/manage/lostBedsController.test.ts

```diff
@@ -0,0 +1,183 @@
+import paths from '../../../paths/temporary-accommodation/manage'
+import { catchValidationErrorOrPropogate, fetchErrorsAndUserInput } from '../../../utils/validation'
+import premisesFactory from '../../../testutils/factories/premises'
+import roomFactory from '../../../testutils/factories/room'
+import { allStatuses } from '../../../utils/lostBedUtils'
+
+jest.mock('../../../utils/restUtils')
+jest.mock('../../../utils/validation')
+
+describe('LostBedsController', () => {
+  const callConfig = { token: 'some-call-config-token' } as CallConfig
+  const premisesId = 'premisesId'
+  const roomId = 'roomId'
+
+  let request: Request
+
+  const response: DeepMocked<Response> = createMock<Response>({})
+  const next: DeepMocked<NextFunction> = createMock<NextFunction>({})
+
+  const lostBedService = createMock<LostBedService>({})
+  const premisesService = createMock<PremisesService>({})
+  const bedspaceService = createMock<BedspaceService>({})
+
+  const lostBedsController = new LostBedsController(lostBedService, premisesService, bedspaceService)
+
+  beforeEach(() => {
+    request = createMock<Request>()
+    ;(extractCallConfig as jest.MockedFn<typeof extractCallConfig>).mockReturnValue(callConfig)
+  })
+
+  describe('new', () => {
+    it('renders the form', async () => {
+      request.params = {
+        premisesId,
+        roomId,
+      }
+
+      const premises = premisesFactory.build()
+      const room = roomFactory.build()
+
+      premisesService.getPremises.mockResolvedValue(premises)
+      bedspaceService.getRoom.mockResolvedValue(room)
+
+      const requestHandler = lostBedsController.new()
+      ;(fetchErrorsAndUserInput as jest.Mock).mockReturnValue({ errors: {}, errorSummary: [], userInput: {} })
+
+      await requestHandler(request, response, next)
+
+      expect(premisesService.getPremises).toHaveBeenCalledWith(callConfig, premisesId)
+      expect(bedspaceService.getRoom).toHaveBeenCalledWith(callConfig, premisesId, roomId)
+
+      expect(response.render).toHaveBeenCalledWith('temporary-accommodation/lost-beds/new', {
+        premises,
+        room,
+        errors: {},
+        errorSummary: [],
+      })
+    })
+
+    describe('create', () => {
+      it('creates a lost bed and redirects to the show lost bed page', async () => {
+        const requestHandler = lostBedsController.create()
+
+        const lostBed = lostBedFactory.build()
+        lostBedService.createLostBed.mockResolvedValue(lostBed)
+
+        request.params = {
+          premisesId,
+          roomId,
+        }
+        request.body = {
+          'startDate-year': 2022,
+          'startDate-month': 8,
+          'startDate-day': 22,
+          'endDate-year': 2022,
+          'endDate-month': 9,
+          'endDate-day': 22,
+          notes: lostBed.notes,
+          reason: lostBed.reason.id,
+          bedId: lostBed.bed.id,
+          referenceNumber: lostBed.referenceNumber,
+          serviceName: 'temporary-accommodation',
```

Comment:
Can you use the newLostBedFactory, and replace this all with something like:
```
...newLostBed,
...DateFormats.convertIsoToDateAndTimeInputs(lostBed.startDate, 'startDate'),
...DateFormats.convertIsoToDateAndTimeInputs(lostBed.endDate, 'endDate'),
```
?